packages/mobile/src/components/Avatar.tsx

```diff
@@ -49,20 +45,51 @@ export function getE164Number(e164Number?: string, recipient?: Recipient) {
 }
 
 export function Avatar(props: Props) {
-  const defaultCountryCode = useSelector(defaultCountryCodeSelector)
-  const { recipient, e164Number, iconSize = DEFAULT_ICON_SIZE, displayNameStyle } = props
+  const defaultCountryCode = useSelector(defaultCountryCodeSelector) ?? undefined
+  const { address, recipient, e164Number, iconSize = DEFAULT_ICON_SIZE, displayNameStyle } = props
+
+  const name = getDisplayName(props)
+  const e164NumberToShow = getE164Number(e164Number, recipient)
+  const thumbnailPath = getRecipientThumbnail(recipient)
 
   return (
-    <BaseAvatar
-      {...props}
-      defaultCountryCode={defaultCountryCode ? defaultCountryCode : undefined}
-      name={getDisplayName(props)}
-      e164Number={getE164Number(e164Number, recipient)}
-      iconSize={iconSize}
-      thumbnailPath={getRecipientThumbnail(recipient)}
-      displayNameStyle={displayNameStyle}
-    />
+    <View style={styles.container}>
+      <ContactCircle thumbnailPath={thumbnailPath} name={name} address={address} size={iconSize} />
+      <Text
+        style={[displayNameStyle || fontStyles.small500, styles.contactName]}
+        numberOfLines={1}
+        ellipsizeMode="tail"
+      >
+        {name}
+      </Text>
+      {e164NumberToShow ? (
+        <PhoneNumberWithFlag
+          e164PhoneNumber={e164NumberToShow}
+          defaultCountryCode={defaultCountryCode}
+        />
+      ) : null}
+
+      {!e164NumberToShow && address ? (
```

Comment:
So if a user has gone through onboarding without entering a phone number, it will show the address near the avatar instead of the phone number with flag? How is it practically possible to be in this state? I thought it was impossible to proceed through onboarding without a phone number?

Reply:
Might not be. I moved this file from the `react-components` because it depended on `ContactCircle` and I had to move `ContactCircle` because I needed access to the redux store. I didn't take a close look at the code but I'll do that and remove what doesn't make sense anymore.

Reply:
I should have noted this on the PR description or as a comment, sorry about that.

Reply:
This is as you said, if a user doesn't enter then phone during onboarding the address is shown. Phone number is no longer required with CIP-8 during onboarding, it's requested on the `VerificationEducationScreen` before starting the verification flow